plugins/tiles/src/main/java/org/apache/struts2/tiles/StrutsApplicationResource.java

```diff
@@ -19,19 +19,47 @@
 package org.apache.struts2.tiles;
 
 import org.apache.tiles.request.locale.PostfixedApplicationResource;
+import org.apache.tiles.request.locale.URLApplicationResource;
 
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
+import java.net.URI;
 import java.net.URL;
 
 public class StrutsApplicationResource extends PostfixedApplicationResource {
 
     private final URL url;
+    private final File file;
+
+    /**
+     * fixes WW-5011
+     * @return file path for "file" protocol elsewhere url path as before to keep backward-compatibility
+     * @see URLApplicationResource#getFile(URL)
+     */
+    private static String getFilePath(URL url) {
+        String path = url.getPath();
+        if (!"file".equals(url.getProtocol())) {
+            return path;
+        }
+        try {
+            // fixes WW-5011 because includes ref in path - like URLApplicationResource#getFile(URL)
+            path = (new URI(url.toExternalForm())).getSchemeSpecificPart();
+        } catch (Exception e) {
+            // fallback solution
+            if (url.getRef() != null && !new File(path).exists()) {
+                // it's like WW-5011
+                path += "#" + url.getRef();
+            }
+        }
+
+        return path;
+    }
 
     public StrutsApplicationResource(URL url) {
-        super(url.getPath());
+        super(getFilePath(url));
```

Comment:
One last suggestion :smile: for the constructor (avoids calling getFilePath twice/regenerating the path string):
```java
final String urlPath = getFilePath(url);
super(urlPath);
this.url = url;
this.file = new File(urlPath);
```